### ISSUE TITLE
CS8168, CS8350 mis-fire when no ref struct applies

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#64441]
**Version Used**: VS 17.4 Preview 3 (32926.485.main)

**Steps to Reproduce**:

1. Create a new C# console application.
2. Set TargetFramework to net7.0 and LangVersion to preview
3. Fill out the Program.cs file with this code:
```cs
class Program
{
    public struct STARTUPINFOW
    {
    }

    public static void CreateProcessW(ref Span<char> commandLine, in STARTUPINFOW startupInfo)
    {
    }

    static void Main()
    {
        var startupInfo = new STARTUPINFOW();
        var args = Span<char>.Empty;

        CreateProcessW(ref args, in startupInfo);
    }
}
```

A customer reported this as a bug against CsWin32 (microsoft/CsWin32#697), but given the docs and error message, I wonder if this is a C# 11 compiler bug. A static method cannot retain a ref/in reference like an instance method on a ref struct can. Why are these errors being produced?

**Expected Behavior**:

No errors.

**Actual Behavior**:

```
1>C:\temp\ConsoleApp1\ConsoleApp1\Program.cs(16,37,16,48): error CS8168: Cannot return local 'startupInfo' by reference because it is not a ref local
1>C:\temp\ConsoleApp1\ConsoleApp1\Program.cs(16,9,16,49): error CS8350: This combination of arguments to 'Program.CreateProcessW(ref Span<char>, in Program.STARTUPINFOW)' is disallowed because it may expose variables referenced by parameter 'startupInfo' outside of their declaration scope
```

**Comment by Sergio0694** [comment]
This is now fixed, and if you try with `4.4.0-3.22479.16` or newer it no longer repros. The issue is that before the introduction of return-only scope, that `startupInfo` reference could've escaped through `commandLine`. That would've caused `args`, which is not scoped, to end up with a reference to `startupInfo`, a local, but with a STE set to caller-method, which would've caused a potential dangling pointer if returned. In that case, the error was correct and by design. The fact that the `struct` is not a `ref stru ct` (I mean `STARTUPINFOW` doesn't matter here).

This is no longer an issue now because the RSTE of `ref`/`in` parameters is now return-only by default, meaning that `startupInfo` in `CreateProcessW` can no longer escape through `commandLine`, hence the method is now safe with no annotations.

As a temporary workaround using .NET 7 RC1, just make `startupInfo` a `scoped in`, that'll also fix it 🙂

**Comment by AArnott** [comment]
Thank you.
